File: TCS Codevita/D minimize sum/D_minimize_sum.py
import math


N, K = input().split()
N = int(N)
K = int(K)


arrList = list(map(int, input().split()))
# print(arrList)

for i in range(1, K+1):
     ind = arrList.index(max(arrList))
     arrList[ind] = math.floor(arrList[ind]/2)
     

# print(arrList)
print(sum(arrList))

File: TCS Codevita/D minimize sum/D_minimize_sum.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    vector<int> arrList;
    int temp;

    cin>>N>>K;
    for(i=0; i<N; i++){
        cin>>temp;
        arrList.push_back(temp);
    }

    for(i=0; i<K; i++){
        maxInd = distance(arrList, max_element(arrList, arrList + arrList.size()));
        arrList[maxInd] = floor(arrList[maxInd]/2);
    }

    cout<<sum(arrList);
    return 0;
}